diff --git a/lib/x86/cpu_features.h b/lib/x86/cpu_features.h
index 8b29816..6b92f84 100644
--- a/lib/x86/cpu_features.h
+++ b/lib/x86/cpu_features.h
@@ -165,7 +165,8 @@ static inline u32 get_x86_cpu_features(void) { return 0; }
 #  define HAVE_AVXVNNI(features)	((features) & X86_CPU_FEATURE_AVXVNNI)
 #endif
 
-#if (GCC_PREREQ(14, 0) || CLANG_PREREQ(18, 0, 18000000)) \
+#if ((GCC_PREREQ(14, 0) && !GCC_PREREQ(16, 0)) || \
+     (CLANG_PREREQ(18, 0, 18000000) && !CLANG_PREREQ(19, 0, 19000000))) \
 	&& !defined(__EVEX512__) /* avoid subtracting the evex512 feature */
 #  define EVEX512	",evex512"	/* needed to override potential -mno-evex512 */
 #  define NO_EVEX512	",no-evex512"
